### ISSUE TITLE
`no-switch-case-fall-through` fails when `continue`ing outer loop

### ISSUE THREAD

**Issue by impinball** [palantir/tslint#509]
The following code fails to check, even though it clearly breaks from the switch-case:

```ts
loop:
while (condition) {
  switch (foo) {
  case bar:
    continue loop; // Line with errors
  default:
  }
}
```

The output I got (with every option except for that one above explicitly disabled):

```
tmp.ts[5, 19]: expected a 'break' before 'case'
tmp.ts[5, 19]: expected a 'break' before 'default'
```

It should be able to infer that it's clearly breaking a parent loop. This is with the current npm CLI version.

**Comment by leeavital** [comment]
````typescript
loop:
while (condition) {
  switch (foo) {
  case bar:
    return true; // Using return instead of break.
  default:
  }
}
````

*Will* pass, so adding continue is consistent with the rule.
